src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -352,6 +352,9 @@ private void initializeReference() {
         if ( referenceURL != null ) {
             referenceSource = new ReferenceMultiSource(gcsOptions, referenceURL, getReferenceWindowFunction());
             referenceDictionary = referenceSource.getReferenceSequenceDictionary(readsHeader != null ? readsHeader.getSequenceDictionary() : null);
+            if (referenceDictionary == null) {
+                throw new UserException.MissingReferenceDictFile(referenceURL);
+            }
```

Comment:
Add an integration test showing that this exception gets thrown for a Spark tool given a fasta input with a missing `.dict` file.